src/control/lib/control/pool.go

```diff
@@ -1252,20 +1252,31 @@ func GetMaxPoolSize(ctx context.Context, log logging.Logger, rpcClient UnaryInvo
 				hostStorageSet.HostSet.String())
 		}
 
-		// FIXME (DAOS-9557) At this time the rank associated to one namespace is not
-		// defined in the StorageScanResp.  Thus we rely on the hypothesis that each SCM
-		// namespace is associated to one and only one rank. Eventually, the protocol should
-		// be changed to define if a SCM namespace is associated with one rank or not. If
-		// yes, it should define with which rank the SCM namespace is associated.
 		for _, scmNamespace := range hostStorage.ScmNamespaces {
 			if scmNamespace.Mount == nil {
+				log.Debugf("Skipping SCM device %s (bdev %s, name %s)",
+					scmNamespace.UUID, scmNamespace.BlockDevice, scmNamespace.Name)
+				continue
+			}
+
+			if len(ranks) != 0 && !scmNamespace.Mount.Rank.InList(ranks) {
+				log.Debugf("Skipping SCM device %s (bdev %s, name %s, rank %s) not in ranklist: %s",
+					scmNamespace.UUID, scmNamespace.BlockDevice, scmNamespace.Name,
+					scmNamespace.Mount.Rank, ranks)
 				continue
 			}
+
 			scmNamespaceFreeBytes := scmNamespace.Mount.AvailBytes
 
 			if scmBytes > scmNamespaceFreeBytes {
 				scmBytes = scmNamespaceFreeBytes
 			}
+
+			if sn, ok := ranksWithoutSmd[scmNamespace.Mount.Rank]; ok {
+				return 0, 0, errors.Errorf("Rank %s with multiple SCM devices: scm1=%s, scm2=%s",
+					scmNamespace.Mount.Rank.String(), sn.UUID, scmNamespace.UUID)
+			}
+			ranksWithoutSmd[scmNamespace.Mount.Rank] = scmNamespace
```

Comment:
I'm finding the logic here a little difficult to follow. Maybe I'm hung up on the name of the map... Isn't the test here to find ranks that have already been associated with a namespace, and flag a rank associated with > 1 namespace as an error?

Oh, I see that it's used again later on. Maybe this whole section of code could use some comments explaining the overall goal and logic.

Reply:
- [ ] Add some comments explaining the aim of the ranksWithoutSmd variable.